test/test_cotrip.py

```diff
@@ -0,0 +1,561 @@
+      },
+      "geometry": {
+        "type": "LineString",
+        "coordinates": [
+          [
+            -104.828415,
+            37.735142
+          ],
+          [
+            -104.830933,
+            37.741074
+          ],
+          [
+            -104.83094,
+            37.741085
+          ],
+          [
+            -104.831367,
+            37.7421
+          ],
+          [
+            -104.831383,
+            37.742138
+          ],
+          [
+            -104.831818,
+            37.743164
+          ],
+          [
+            -104.831825,
+            37.743183
+          ],
+          [
+            -104.832626,
+            37.745064
+          ],
+          [
+            -104.832634,
+            37.745079
+          ],
+          [
+            -104.834328,
+            37.749054
+          ]
+        ]
+      }
+    }
+
+    assert test_feature == expected_feature
+
+def test_parse_alert_no_data():
+  test_feature = cotrip_translator.parse_alert(None)
+  expected_feature=None
+  assert test_feature == expected_feature
+
+def test_parse_alert_invalid_data(): 
+  test_var = 'a,b,c,d'
+  callback = MagicMock()
+  test_feature = cotrip_translator.parse_alert(test_var, callback_function=callback)
+  assert callback.called and test_feature == None
+  
+#--------------------------------------------------------------------------------Unit test for validate_ function--------------------------------------------------------------------------------
+def test_validate_alert_valid_data():
+  test_valid_output = {
+    "rtdh_timestamp": 1615866698.393723,
+    "rtdh_message_id": "dd962abd-0afa-4810-aac0-165edb834e71",
+    "event": {
+        "type": "Construction",
+        "sub_type": "Work Zone:roadway-relocation",
+        "source": {
+            "id": "349611",
+            "type": "Road Work",
+            "sub_type": "Road Construction",
+            "collection_timestamp": 1615482720
+        },
+        "lrs": None,
+        "geometry": "LINESTRING (-104.48011 37.007645, -104.480103 37.008034, -104.480125 37.008469, -104.480202 37.008904, -104.48024 37.009048, -104.480324 37.009338, -104.482475 37.015327, -104.482712 37.015945, -104.48288 37.016335, -104.482979 37.016521, -104.483208 37.016884, -104.483467 37.01722, -104.483612 37.01738, -104.483925 37.017681, -104.484253 37.017948, -104.484772 37.018295, -104.485138 37.01849, -104.485504 37.018661, -104.485886 37.01881, -104.486473 37.019005, -104.488014 37.019493)",
+        "header": {
+            "description": "Road Construction - I-25 (Main St.) business loop from MP 1-2",
+            "start_timestamp": 1615813200,
+            "end_timestamp": 1638255600
+        },
+        "detail": {       
+            "road_name": "I-25",
+            "road_number": "I-25",
+            "direction": "North"
+        },
+    }
+}
+  assert cotrip_translator.validate_alert(test_valid_output) == True
+
+def test_validate_incident_missing_required_field_description():
```

Comment:
Should be named validate alert, not incident.

Reply:
changed it to test_validate_alert_missing_required_field_description